src/Admin/AdminExtensionInterface.php

```diff
@@ -20,6 +20,12 @@
 use Sonata\AdminBundle\Show\ShowMapper;
 use Sonata\CoreBundle\Validator\ErrorElement;
 
+@trigger_error(sprintf(
+    '"%s" is deprecated since 3.x and will be removed in 4.0.'
+            .'Implement single interfeces from "Sonata\AdminBundle\Admin\Extension" namespace.',
+    AdminExtensionInterface::class
+), E_USER_DEPRECATED);
+
```

Comment:
This interface should stay, shouldn't be deprecated, and should extend the other ones, like here: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Admin/AdminInterface.php#L33

Reply:
but why? If we separate all the methods, why do we need this one?

Reply:
Also what about abstract extension class? I think that one should also be deprecated

Reply:
We need it because of https://github.com/sonata-project/SonataAdminBundle/pull/5164#issuecomment-407546988

Reply:
that is why I added a new empty Interface`Sonata\AdminBundle\Admin\Extension\ AdminExtensionInterface`, that can be used for addExtension but changing it would be a BC Break.